library/std/src/sys/unix/process/process_common.rs

```diff
@@ -43,17 +44,56 @@ cfg_if::cfg_if! {
             set.write_bytes(0u8, 1);
             return 0;
         }
+        pub unsafe fn sigfillset(set: *mut libc::sigset_t) -> libc::c_int {
+            set.write_bytes(0xffu8, 1);
+            return 0;
+        }
         #[allow(dead_code)]
         pub unsafe fn sigaddset(set: *mut libc::sigset_t, signum: libc::c_int) -> libc::c_int {
             use crate::{slice, mem};
+            let size = mem::size_of::<libc::sigset_t>();
 
-            let raw = slice::from_raw_parts_mut(set as *mut u8, mem::size_of::<libc::sigset_t>());
             let bit = (signum - 1) as usize;
+            if bit < 0 || bit >= 8 * size {
+                set_errno(libc::EINVAL);
+                return -1;
+            }
+
+            let raw = slice::from_raw_parts_mut(set as *mut u8, size);
             raw[bit / 8] |= 1 << (bit % 8);
```

Comment:
Hm, this isn't your fault, but these impls don't match the cited [android impls](https://github.com/aosp-mirror/platform_bionic/blob/ad8dcd6023294b646e5a8288c0ed431b0845da49/libc/include/android/legacy_signal_inlines.h). In particular, it uses `c_long` rather than u8. (`LONG_BIT` is `size_of::<c_long>() * 8`, but we just are using `8`)

I think this will result in incorrect behavior on big endian android, although it's not hard to guess why nobody's noticed before.

Reply:
Thanks for catching this! Once #100782 lands I'll rebase and port the changes there over here. (Also I noticed you fixed up the error handling in that PR; thanks for doing that!)